medusa/tv.py

```diff
@@ -195,13 +195,33 @@ def __init__(self, indexer, indexerid, lang='', quality=None, flatten_folders=No
         self.release_groups = None
         self.exceptions = []
         self.externals = {}
+        self._cached_indexer_api = None
 
         other_show = Show.find(app.showList, self.indexerid)
         if other_show is not None:
             raise MultipleShowObjectsException("Can't create a show if it already exists")
 
         self._load_from_db()
 
+    def get_indexer_api(self, indexer_api=None, refresh=False):
```

Comment:
This method is unpythonic.  Having a getter that serves dual purpose as a setter is poor practice.